File: C++/Q37.cpp
#include <iostream>
using namespace std;

int gcd(int a, int b) {
	if (b == 0) return a >= 0 ? a : -a;
	return gcd(b, a % b);
}

int main() {
	int a, b;
	cout << "Enter two integers: ";
	if (!(cin >> a >> b)) { cout << "Invalid input." << endl; return 1; }
	cout << "GCD(" << a << ", " << b << ") = " << gcd(a, b) << endl;
	return 0;
}

File: Java/Q37.java
import java.util.Scanner;

public class GCDRecursion {
    static int gcd(int a, int b) {
        if (b == 0)
            return a;
        return gcd(b, a % b);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter two numbers: ");
        int a = sc.nextInt();
        int b = sc.nextInt();
        System.out.println("GCD of " + a + " and " + b + " = " + gcd(a, b));
        sc.close();
    }
}
